src/main/java/bio/terra/workspace/service/job/JobService.java

```diff
@@ -0,0 +1,381 @@
+          stairwayJdbcConfiguration.isForceClean(),
+          stairwayJdbcConfiguration.isMigrateUpgrade());
+
+    } catch (StairwayException stairwayEx) {
+      throw new InternalStairwayException("Stairway initialization failed", stairwayEx);
+    }
+  }
+
+  public void releaseJob(String jobId, AuthenticatedUserRequest userReq) {
+    try {
+      if (userReq != null) {
+        // currently, this check will be true for stewards only
+        boolean canDeleteAnyJob =
+            samService.isAuthorized(
+                userReq.getRequiredToken(),
+                SamUtils.SAM_WORKSPACE_MANAGER_RESOURCE,
+                appConfig.getResourceId(),
+                SamUtils.SAM_WORKSPACE_MANAGER_DELETE_JOBS_ACTION);
+
+        // if the user has access to all jobs, no need to check for this one individually
+        // otherwise, check that the user has access to this job before deleting
+        if (!canDeleteAnyJob) {
+          verifyUserAccess(jobId, userReq); // jobId=flightId
+        }
+      }
+      stairway.deleteFlight(jobId, false);
+    } catch (StairwayException stairwayEx) {
+      throw new InternalStairwayException(stairwayEx);
+    } catch (ApiException samEx) {
+      throw new SamApiException(samEx);
+    }
+  }
+
+  public JobModel mapFlightStateToJobModel(FlightState flightState) {
+    FlightMap inputParameters = flightState.getInputParameters();
+    String description = inputParameters.get(JobMapKeys.DESCRIPTION.getKeyName(), String.class);
+    FlightStatus flightStatus = flightState.getFlightStatus();
+    String submittedDate = flightState.getSubmitted().toString();
+    JobModel.StatusEnum jobStatus = getJobStatus(flightStatus);
+
+    String completedDate = null;
+    HttpStatus statusCode = HttpStatus.ACCEPTED;
+
+    if (flightState.getCompleted().isPresent()) {
+      FlightMap resultMap = getResultMap(flightState);
+      // The STATUS_CODE return only needs to be used to return alternate success responses.
+      // If it is not present, then we set it to the default OK status.
+      statusCode = resultMap.get(JobMapKeys.STATUS_CODE.getKeyName(), HttpStatus.class);
+      if (statusCode == null) {
+        statusCode = HttpStatus.OK;
+      }
+
+      completedDate = flightState.getCompleted().get().toString();
+    }
+
+    JobModel jobModel =
+        new JobModel()
+            .id(flightState.getFlightId())
+            .description(description)
+            .status(jobStatus)
+            .statusCode(statusCode.value())
+            .submitted(submittedDate)
+            .completed(completedDate);
+
+    return jobModel;
+  }
+
+  private JobModel.StatusEnum getJobStatus(FlightStatus flightStatus) {
+    switch (flightStatus) {
+      case ERROR:
+      case FATAL:
+        return JobModel.StatusEnum.FAILED;
+      case RUNNING:
+        return JobModel.StatusEnum.RUNNING;
+      case SUCCESS:
+        return JobModel.StatusEnum.SUCCEEDED;
+    }
+    return JobModel.StatusEnum.FAILED;
+  }
```

Comment:
Not sure I follow - do you mean using the stairyway.FlightStatus as the field type for the generated JobModel's status?
I'd rather have the JobModel status defined in the same API that its used, rather than linking Stairway to this service more tightly. But, not sure I totally understand the comment.

Reply:
that's what I meant - instead of returning a  `JobModel.StatusEnum`, return a `bio.terra.stairway.FlightStatus` directly. I don't feel strongly about this, and you have a reason for doing it the way you did, so :+1: